[GTK][WPE] Add GLDisplay wrapper

carlosgcampos opened the pull request:
#### 04dd368b15065ab5dad288e2d5043a052a35591e
<pre>
[GTK][WPE] Add GLDisplay wrapper
<a href="https://bugs.webkit.org/show_bug.cgi?id=278227">https://bugs.webkit.org/show_bug.cgi?id=278227</a>

Reviewed by Adrian Perez de Castro.

Similar to GLContext but for EGLDisplay. For now it will be used
internally by PlatformDisplay, but the plan is to use it from gtk
specific code in the UI process once we remove the remaining gtk code
from PlatformDisplay.

* Source/WebCore/Headers.cmake:
* Source/WebCore/PlatformPlayStation.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::PlatformDisplay):
(WebCore::PlatformDisplay::~PlatformDisplay):
(WebCore::PlatformDisplay::eglDisplay const):
(WebCore::PlatformDisplay::eglCheckVersion const):
(WebCore::PlatformDisplay::eglExtensions const):
(WebCore::PlatformDisplay::initializeEGLDisplay):
(WebCore::PlatformDisplay::terminateEGLDisplay):
(WebCore::PlatformDisplay::createEGLImage const):
(WebCore::PlatformDisplay::destroyEGLImage const):
(WebCore::PlatformDisplay::dmabufFormats):
* Source/WebCore/platform/graphics/PlatformDisplay.h:
(WebCore::PlatformDisplay::gtkEGLDisplay):
* Source/WebCore/platform/graphics/egl/GLDisplay.cpp: Added.
(WebCore::GLDisplay::create):
(WebCore::GLDisplay::GLDisplay):
(WebCore::GLDisplay::terminate):
(WebCore::GLDisplay::checkVersion const):
(WebCore::GLDisplay::createImage const):
(WebCore::GLDisplay::destroyImage const):
(WebCore::GLDisplay::dmabufFormats):
* Source/WebCore/platform/graphics/egl/GLDisplay.h: Added.
(WebCore::GLDisplay::eglDisplay const):
(WebCore::GLDisplay::extensions const):
* Source/WebCore/platform/graphics/egl/PlatformDisplaySurfaceless.cpp:
(WebCore::PlatformDisplaySurfaceless::PlatformDisplaySurfaceless):
* Source/WebCore/platform/graphics/gbm/PlatformDisplayGBM.cpp:
(WebCore::PlatformDisplayGBM::PlatformDisplayGBM):
* Source/WebCore/platform/graphics/libwpe/PlatformDisplayLibWPE.cpp:
(WebCore::PlatformDisplayLibWPE::PlatformDisplayLibWPE):
(WebCore::PlatformDisplayLibWPE::initializeEGLDisplay):
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.cpp:
(WebCore::PlatformDisplayWayland::gtkEGLDisplay):
(WebCore::PlatformDisplayWayland::initializeEGLDisplay):
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.h:
* Source/WebCore/platform/graphics/win/PlatformDisplayWin.cpp:
(WebCore::PlatformDisplayWin::initializeEGLDisplay):
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp:
(WebCore::PlatformDisplayX11::gtkEGLDisplay):
(WebCore::PlatformDisplayX11::initializeEGLDisplay):
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.h:

Canonical link: <a href="https://commits.webkit.org/282388@main">https://commits.webkit.org/282388@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e2c7eba6bf3c1f001f1c1f786f679477c7e7bd5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/62898 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42254 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15494 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/66919 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13502 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65018 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/49941 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/13786 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50715 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9323 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/65967 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39273 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54482 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31397 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/35962 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/11815 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12378 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57510 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12145 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/68614 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/6844 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/11774 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58029 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/6876 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54542 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58226 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/5716 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9500 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38074 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39154 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40265 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/38896 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebCore/platform/graphics/PlatformDisplay.cpp:
```diff
@@ -244,67 +229,36 @@ EGLDisplay PlatformDisplay::eglDisplay() const
     if (!m_eglDisplayInitialized)
         const_cast<PlatformDisplay*>(this)->initializeEGLDisplay();
-    return m_eglExtensions;
+    static GLDisplay::Extensions empty;
```
This can be `static const`:

```suggestion
    static const GLDisplay::Extensions empty;
```

carlosgcampos replied:
This will be removed in the next patch.

aperezdc commented on Source/WebCore/platform/graphics/PlatformDisplay.cpp:
```diff
@@ -339,108 +293,29 @@ void PlatformDisplay::terminateEGLDisplay()
-        });
-    });
-    return m_dmabufFormats;
+    static Vector<GLDisplay::DMABufFormat> empty;
```
Same here:

```suggestion
    static const Vector<GLDisplay::DMABufFormat> empty;
```

carlosgcampos replied:
And this too. I'll change it anyway before landing

aperezdc reviewed (approved):
LGTM, with just two very minor comments in case you may want to check before landing.

carlosgcampos committed:
[GTK][WPE] Add GLDisplay wrapper
https://bugs.webkit.org/show_bug.cgi?id=278227

Reviewed by Adrian Perez de Castro.

Similar to GLContext but for EGLDisplay. For now it will be used
internally by PlatformDisplay, but the plan is to use it from gtk
specific code in the UI process once we remove the remaining gtk code
from PlatformDisplay.

* Source/WebCore/Headers.cmake:
* Source/WebCore/PlatformPlayStation.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/graphics/PlatformDisplay.cpp:
(WebCore::PlatformDisplay::PlatformDisplay):
(WebCore::PlatformDisplay::~PlatformDisplay):
(WebCore::PlatformDisplay::eglDisplay const):
(WebCore::PlatformDisplay::eglCheckVersion const):
(WebCore::PlatformDisplay::eglExtensions const):
(WebCore::PlatformDisplay::initializeEGLDisplay):
(WebCore::PlatformDisplay::terminateEGLDisplay):
(WebCore::PlatformDisplay::createEGLImage const):
(WebCore::PlatformDisplay::destroyEGLImage const):
(WebCore::PlatformDisplay::dmabufFormats):
* Source/WebCore/platform/graphics/PlatformDisplay.h:
(WebCore::PlatformDisplay::gtkEGLDisplay):
* Source/WebCore/platform/graphics/egl/GLDisplay.cpp: Added.
(WebCore::GLDisplay::create):
(WebCore::GLDisplay::GLDisplay):
(WebCore::GLDisplay::terminate):
(WebCore::GLDisplay::checkVersion const):
(WebCore::GLDisplay::createImage const):
(WebCore::GLDisplay::destroyImage const):
(WebCore::GLDisplay::dmabufFormats):
* Source/WebCore/platform/graphics/egl/GLDisplay.h: Added.
(WebCore::GLDisplay::eglDisplay const):
(WebCore::GLDisplay::extensions const):
* Source/WebCore/platform/graphics/egl/PlatformDisplaySurfaceless.cpp:
(WebCore::PlatformDisplaySurfaceless::PlatformDisplaySurfaceless):
* Source/WebCore/platform/graphics/gbm/PlatformDisplayGBM.cpp:
(WebCore::PlatformDisplayGBM::PlatformDisplayGBM):
* Source/WebCore/platform/graphics/libwpe/PlatformDisplayLibWPE.cpp:
(WebCore::PlatformDisplayLibWPE::PlatformDisplayLibWPE):
(WebCore::PlatformDisplayLibWPE::initializeEGLDisplay):
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.cpp:
(WebCore::PlatformDisplayWayland::gtkEGLDisplay):
(WebCore::PlatformDisplayWayland::initializeEGLDisplay):
* Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.h:
* Source/WebCore/platform/graphics/win/PlatformDisplayWin.cpp:
(WebCore::PlatformDisplayWin::initializeEGLDisplay):
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.cpp:
(WebCore::PlatformDisplayX11::gtkEGLDisplay):
(WebCore::PlatformDisplayX11::initializeEGLDisplay):
* Source/WebCore/platform/graphics/x11/PlatformDisplayX11.h:

Canonical link: https://commits.webkit.org/282388@main
